CONTRIBUTING.md

```diff
@@ -1,3 +1,93 @@
+-  If you'd like to add new functionality to this project, describe the problem you want to solve in a [new Issue](https://github.com/forcedotcom/run-code-analyzer/issues/new).
+-  Issues that have been identified as a feature request will be labelled `enhancement`.
+-  If you'd like to implement the new feature, please wait for feedback from the project
+   maintainers before spending too much time writing the code. In some cases, `enhancement`s may
+   not align well with the project objectives at the time.
+
+### Tests, Documentation, Miscellaneous
+-  If you'd like to improve the tests, you want to make the documentation clearer, you have an
+   alternative implementation of something that may have advantages over the way its currently
+   done, or you have any other change, we would be happy to hear about it!
+-  If its a trivial change, go ahead and [send a Pull Request](#creating_a_pull_request) with the changes you have in mind.
+-  If not, [open an Issue](https://github.com/forcedotcom/run-code-analyzer/issues/new) to discuss the idea first.
+
+# Contribution Checklist
+
+- [x] Clean, simple, well styled code
+- [x] Commits should be atomic and messages must be descriptive. Related issues should be mentioned by Issue number.
+- [x] Comments
+    - Module-level & function-level comments.
+    - Comments on complex blocks of code or algorithms (include references to sources).
+- [x] Tests
+    - The test suite, if provided, must be complete and pass
+    - Increase code coverage, not versa.
+- [x] Dependencies
+    - Minimize number of dependencies.
+    - Prefer Apache 2.0, BSD3, MIT, ISC and MPL licenses.
+- [x] Reviews
+    - Changes must be approved via peer code review
+
+# Local development
+
+## Install dependencies
+Install required node dependencies:
+```
+npm install
+```
+
+## Running tests
+We strive for 100% code coverage.
+After making your changes, run your tests and check coverage with:
+```
+npm run test
+```
+
+## Before submitting
+We use `ncc` to produce a single *index.js* file inside the *dist* directory from all the typescript source code.
+Therefore, before you can submit, you must create this package along with license files, and badges with:
+```
+npm run all
+```
+
+# Creating a Pull Request
+
+1. **Ensure the bug/feature was not already reported** by searching on GitHub under Issues.  If none exists, create a new issue so that other contributors can keep track of what you are trying to add/fix and offer suggestions (or let you know if there is already an effort in progress).
+3. **Clone** the forked repo to your machine.
+4. **Create** a new branch to contain your work (e.g. `git br fix-issue-11`)
+4. **Commit** changes to your own branch.
+5. **Push** your work back up to your fork. (e.g. `git push fix-issue-11`)
+6. **Submit** a Pull Request against the `main` branch and refer to the issue(s) you are fixing. Try not to pollute your pull request with unintended changes. Keep it simple and small.
```

Comment:
Submit a pull request against the `main` branch and refer to the issue you're fixing. Keep your pull request simple and small to avoid any unintended changes.

(similar comments as previously - no caps on pull request, use contractions, plus simplified last two sentences)

Reply:
This is directly from the already approved oss template: https://github.com/salesforce/oss-template/blob/main/CONTRIBUTING.md

Do we really want to edit what they have over there?

Reply:
Updated with your suggestions.